lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -803,6 +771,90 @@ defmodule Thrift.Generator.StructBinaryProtocol do
     |> list_deserializer(name, file_group)
   end
 
+  def struct_serializer(%Union{fields: fields}, name, file_group) do
+    fields = Enum.reject(fields, &(&1.type == :void))
+
+    single_field_serializers = Enum.map(fields, fn
+      %Field{name: match_name} = match_field ->
+        nil_matchers = Enum.flat_map(fields, fn
```

Comment:
How does this compare performance-wise to:

```elixir
nil_matchers = Enum.filter_map(fields,
  fn(%Field{name: match_name}) -> match_name != name end,
  fn(%Field{name: match_name}) -> {match_name, nil} end)
```

I imagine your approach is faster because it can match directly on the name, but `Enum.filter_map/3` uses fewer intermediate lists.

This is just the codegen bit, though, so readability could trump performance.